include/boost/math/tools/polynomial.hpp

```diff
@@ -98,8 +111,86 @@ T evaluate_chebyshev(const Seq& a, const T& x)
    return a[0] / 2 + yk * x - yk1;
 }
 
+
+template <typename T>
+class polynomial;
+
+
+template <typename T>
+std::pair< polynomial<T>, polynomial<T> >
+unchecked_synthetic_division(const polynomial<T>& dividend, const polynomial<T>& divisor)
+{
+    BOOST_ASSERT(divisor.degree() <= dividend.degree());
+    BOOST_ASSERT(divisor != zero_element(std::multiplies< polynomial<T> >()));
+    BOOST_ASSERT(dividend != zero_element(std::multiplies< polynomial<T> >()));
+    
+    using namespace boost::lambda;
+    
+    std::vector<T> intermediate_result(dividend.data());
+    if (divisor.degree() == T(0))
+        intermediate_result.insert(begin(intermediate_result), T(0));
+    
+    {
+        typedef BOOST_DEDUCED_TYPENAME std::vector<T>::reverse_iterator reverse_iterator;
+        T const normalizer = divisor[divisor.size() - 1];
+        reverse_iterator const last = intermediate_result.rbegin() + dividend.degree() - divisor.degree() + 1;
+        for (reverse_iterator i = intermediate_result.rbegin(); i != last; i++)
+        {
+            if (*i != T(0))
+            {
+                T const coefficient = *i /= normalizer;
+                // for (std::size_t j = 1; j < divisor.degree() + 1; j++)
+                    // *(i + j) -= *(divisor.data().rbegin() + j) * coefficient;
+                reverse_iterator const j = i + 1;
+                std::transform(j, j + divisor.degree(), divisor.data().rbegin() + 1, j, _1 - _2 * coefficient);
```

Comment:
> If it's OK with you I'd rather keep it simple and not broach the 
> question of C++1y at the moment because it opens so many questions. 
> If/when we come to proposing a Boost.Polynomial library in its own 
> right then I would definitely be in favour of C++11 as a minimum or base.
>

That's fine, John.